### PR TITLE
Ore generation

### PR DESCRIPTION
**Pull request by LetterN**
adds the orgen system, does NOT add any ore rather than tin. Will be adding more later if required

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,17 +1,18 @@
 buildscript {
   repositories {
     maven {
-      name 'forge'
-      url 'https://files.minecraftforge.net/maven'
+      // Forge (this is the new url)
```
Comment (josephtarbit) [inline build.gradle]: Remove the comment, re-add the repo name

File: build.gradle [hunk 2]
```diff
@@ -1,17 +1,18 @@
...
+      url 'https://maven.minecraftforge.net'
     }
+    mavenCentral()
     maven {
-      name 'sponge'
+      // Mixin
```
Comment (josephtarbit) [inline build.gradle]: Remove comment, re-add the sponge repo name

File: build.gradle [hunk 3]
```diff
@@ -21,12 +22,13 @@ buildscript {
...
-apply plugin: 'org.spongepowered.mixin'
-apply plugin: 'net.covers1624.signing'
+
 apply plugin: 'eclipse'
 apply plugin: 'maven-publish'
+apply plugin: 'org.spongepowered.mixin'
+apply plugin: 'net.covers1624.signing'
 
-apply from: 'https://raw.githubusercontent.com/SizableShrimp/Forge-Class-Remapper/main/classremapper.gradle'
+// apply from: 'https://raw.githubusercontent.com/SizableShrimp/Forge-Class-Remapper/main/classremapper.gradle'
```
Comment (josephtarbit) [inline build.gradle]: Just completely remove this, forgot to remove it myself

File: build.gradle [hunk 4]
```diff
@@ -155,23 +162,31 @@ repositories {
 }
 
 dependencies {
-    minecraft group: 'net.minecraftforge', name: 'forge', version: "${project.minecraft_version}-${project.forge_version}"
```
Comment (josephtarbit) [inline build.gradle:162]: Add this back

File: src/main/java/mods/railcraft/Railcraft.java [hunk 5]
```diff
@@ -130,8 +144,42 @@ public static Railcraft getInstance() {
...
+        logger.error("RC Commit: " + mf.getValue("Build-Commit"));
+        logger.error("RC Source: PR");
+        jar.close();
+        return;
+      }
+      logger.info("RC Commit: " + mf.getValue("Build-Commit"));
+      logger.info("RC Source: " + (gsource != null ?  gsource : "Localy built."));
+      jar.close();
+    }
+    catch (IOException e){
```
Comment (josephtarbit) [inline src/main/java/mods/railcraft/Railcraft.java]: Move this up onto the same line as the close bracket (like with if/else)

File: src/main/java/mods/railcraft/Railcraft.java [hunk 6]
```diff
@@ -146,8 +194,22 @@ private void handleGatherData(GatherDataEvent event) {
...
+    if (event.getName() != null && RailcraftOrePlacements.TIN_ORE.bannedBiomesList().contains(event.getName().toString())) {
+      return;
+    }
+    event.getGeneration().addFeature(
+        GenerationStep.Decoration.UNDERGROUND_ORES,
+        RailcraftOrePlacements.TIN_ORE.placedFeature());
+  }
+
   @SubscribeEvent
-  public void handleWorldTick(TickEvent.WorldTickEvent event) {
```
Comment (josephtarbit) [inline src/main/java/mods/railcraft/Railcraft.java]: Add this back (maybe change your formatter settings if this was done by your IDE)

File: src/main/java/mods/railcraft/data/worldgen/RailcraftConfigurableOre.java [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+        .defineList("biomeIgnoreList", List.<String>of(),
+            obj -> ResourceLocation.isValidResourceLocation(obj.toString()));
+    builder.pop();
+  }
+
+  /**
+   * Hook for reloading ore configs.
+   */
+  public void handleConfigSpecLoad() {
+    this.placementType = this.placementTypeConfig.get();
```
Comment (josephtarbit) [inline src/main/java/mods/railcraft/data/worldgen/RailcraftConfigurableOre.java]: Why are you copying these values? (Why can't you just reference the config directly)

File: src/main/java/mods/railcraft/world/level/block/RailcraftBlocks.java [hunk 8]
```diff
@@ -1008,6 +1009,34 @@
             BlockBehaviour.Properties.of(Material.WATER)
                 .noCollission().strength(50.0F).noDrops()));
 
+  /* == ORES == */
+
+  // UNUSED! ONLY FOR 1.16 BACKPORT
```
Comment (josephtarbit) [inline src/main/java/mods/railcraft/world/level/block/RailcraftBlocks.java]: Not planning on doing back ports at this point, just remove

**Review by josephtarbit — CHANGES_REQUESTED** [review]

**Comment by LetterN** [comment]
ded pr

**Pull request closed without merging**